README.md

```diff
@@ -0,0 +1,108 @@
+# Introduction
+
+The **hal_gigadevice** is a set of both standard firmware library and ARM CMSIS
+configurations for GigaDevice MCUs. The HAL is organized following the below
+directory structure.
+
+## Directory Structure
+
+The directory is composed by three parts:
+
+ - SoC specific libraries.
+ - ZephyrRTOS module directory.
+ - This README file.
+
+### ARM
+
+Each ARM firmware library is organized in the following structure:
+
+```
+.
+└── gd32xxx
+    ├── cmsis
+    │   └── gd
+    │       └── gd32xxx
+    └── standard_peripheral
+        ├── include
+        └── source
+```
+
+### RISC-V
+
+The RISC-V structure is not defined yet. The riscv sub-directory will change
+based on how [NMSIS library](https://github.com/Nuclei-Software/NMSIS/) will
+be implemented in Zephyr. At present date, the structure only reflects current
+manufacturer version.
+
+```
+└── gd32bf103
+    ├── riscv
+    │   ├── drivers
+    │   ├── env_eclipse
+    │   ├── env_iar
+    │   └── stubs
+    └── standard_peripheral
+        ├── include
+        └── source
+```
+
+
+ Any contribution should follow the `How to submit code` using as premisses the
+ [Zephyr Contribution Guidelines](https://docs.zephyrproject.org/latest/contribute/index.html).
+
+# How to submit code
+
+ - When submiting an updated firmware library version, it is important make
+   sure that the last firmware version will be merged. The library version can
+   be checked at [GigaDevice MCU Site](http://www.gd32mcu.com/en/download/).
+ - This repository is managed mainly on Linux. This means that changes should
+   be submited using Linux LF format. Usually running a similar command to
+   `find * type f -print0 | xargs -0 dos2unix -k` should be enough.
+ - Make sure directory structure is respected.
+ - Directory names should be converted to lowercase.
+ - Run `gd32_api_check` script to make sure gd32 public API is conformance.
+   Adjusts should be made otherwise code can be reject. Exceptions should be
+   discussed at review phase.
+
+## The gd32 standard peripheral API
+
+The gd32 standard peripheral API was changed to avoid that macros, enum values
+and function names colide with content from both ZephyrRTOS core and libraries.
+
+The are two general rules:
+
+ - Public API funtions must be prefixed with `gd32_`.
+ - Public defines and enum values must be uppercase and prefixed with `GD32_`.
+
+The cmsis and riscv contents are not part of standard peripheral library and
+changes are treated as exceptions. This means that `<firmware_library>.h>` and
+`system_<firmware_library>.h>` files should reside on that places.
```

Comment:
`on the same place`?

Reply:
This is because of _The gd32vf103 firmware library have system files at standard peripheral
directory. This move system files to right place at riscv system directory._